request.go

```diff
@@ -212,66 +222,111 @@ func (exo *Client) parseResponse(resp *http.Response) (json.RawMessage, error) {
 	return b, nil
 }
 
-// AsyncRequest performs an asynchronous request and polls it for retries * day [s]
-func (exo *Client) AsyncRequest(req AsyncCommand, async AsyncInfo) (interface{}, error) {
-	body, err := exo.request(req.name(), req)
+func (exo *Client) processAsyncJob(ctx context.Context, job *asyncJob) {
+	defer close(job.responseChan)
+	defer close(job.errorChan)
+
+	body, err := exo.request(job.command.name(), job.command)
```

Comment:
To be complete (but maybe later), you should also have a `exo.requestWithContext(ctx, ...)`.